omnipaxos/src/util.rs

```diff
@@ -16,6 +16,7 @@ pub(crate) struct AcceptedMetaData<T: Entry> {
 #[derive(Debug, Clone, Default)]
 /// Promise without the suffix
 pub(crate) struct PromiseMetaData {
+    pub n: Ballot,
```

Comment:
We didn't store the promise here because the leaderstate is supposed to be cleaned when becoming leader in a new round, and all the `PromiseMetaData` are specific to this round.

Since we right now don't really do anything with nodes with a higher promise (other than not sending prepare/accept to them), I prefer if we instead keep track of them in [`promises_meta` field](https://github.com/haraldng/omnipaxos/blob/ff063b90228d6e1767c9712538bfa4c91bdc2c1e/omnipaxos/src/util.rs#L65) in `LeaderState` where `promises_meta` field is a `Vec<PromiseState>`:

```rust
enum PromiseState {
        NotPromised,
        Promised(PromiseMeta),
        PromisedHigher
}
```

I think that will make the code cleaner and less prone to introducing bugs, we don't store a bunch of ballots we don't use, and we can extend it in the future if we want to do something with those `PromisedHigher`